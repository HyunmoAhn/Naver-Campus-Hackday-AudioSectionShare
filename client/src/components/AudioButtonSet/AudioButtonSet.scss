.AudioButtonSet__play-btn {
  background-color: transparent;
  border: 2px solid #BABABA;
  border-radius: 50%;
  color: white;
  width: 35px;
  height: 35px;
  margin-top: 10px;
  font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
}

.AudioButtonSet__section-btn {
  position: relative;
  background-color: transparent;
  border: 0;
  color: white;
  left: -15px;

  .active {
    color: #66D36E;
  }
}

.AudioButtonSet__toggle-btn {
  position: relative;
  background-color: transparent;
  border: 0;
  color: white;
  left: 15px;

  .active {
    color: #66D36E;
  }
}

.AudioButtonSet__share-btn {
  position: relative;
  width: 100px;
  margin-left: -100px;
  background-color: transparent;
  border: 0;
  left: 150px;
  color: white;

  .active {
    color: #66D36E;
  }
}

.AudioButtonSet__section-loop-cancel {
  background-color: transparent;
  border: 1px solid white;
  color: white;
  position: relative;
  width: 140px;
  margin-right: -140px;
  left: -200px;
  top: -10px;
  height: 35px;
  margin-bottom: -35px;
}