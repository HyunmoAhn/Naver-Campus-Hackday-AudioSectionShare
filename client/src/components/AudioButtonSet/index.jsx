import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import TimeScreen from '../TimeScreen';
import './AudioButtonSet.scss';

const propTypes = {
	audio: PropTypes.object,
	endTime: PropTypes.number,
	isSectionLoop: PropTypes.bool,
	isSetSection: PropTypes.bool,
	isShare: PropTypes.bool,
	startTime: PropTypes.number,
	onSectionLoopCancel: PropTypes.func,
	onSetSection: PropTypes.func,
	onShareSetting: PropTypes.func,
	onToggleLoop: PropTypes.func,
	onTogglePlay: PropTypes.func,
};

const defaultProps = {
	audio: {},
	endTime: 0,
	isSectionLoop: false,
	isSetSection: false,
	isShare: false,
	startTime: 0,
	onSectionLoopCancel() {},
	onSetSection() {},
	onShareSetting() {},
	onToggleLoop() {},
	onTogglePlay() {},
};

class AudioButtonSet extends React.Component {
	render() {
		const { endTime, isSectionLoop, isSetSection, isShare, startTime } = this.props;
		const playBtnClassName = cx('fa', {
			'fa-play': this.props.audio.paused,
			'fa-pause': !this.props.audio.paused,
		});
		const loopBtnClassName = cx('fa', 'fa-retweet', {
			'active': this.props.audio.loop,
		});
		const sectionBtnClassName = cx('fa', 'fa-exchange', {
			'active': isSetSection,
		});

		return (
			<div className="AudioButtonSet">
				{isSectionLoop &&
				<button
					className="AudioButtonSet__section-loop-cancel"
					type="button"
					onClick={this.props.onSectionLoopCancel}
				>
					<TimeScreen second={startTime} />
					~
					<TimeScreen second={endTime} />
					<br />
					반복 해제
				</button>
				}
				<button
					className="AudioButtonSet__section-btn"
					disabled={isSectionLoop}
					type="button"
					onClick={this.props.onSetSection}
				>
					<i className={sectionBtnClassName} />
				</button>
				<button
					className="AudioButtonSet__play-btn"
					type="button"
					onClick={this.props.onTogglePlay}
				>
					<i className={playBtnClassName} />
				</button>
				<button
					className="AudioButtonSet__toggle-btn"
					type="button"
					onClick={this.props.onToggleLoop}
				>
					<i className={loopBtnClassName} />
				</button>
				<button
					className="AudioButtonSet__share-btn"
					type="button"
					onClick={this.props.onShareSetting}
				>
					<span className={cx({ 'active': isShare })}>SNS 공유하기</span>
				</button>
			</div>
		);
	}
}

AudioButtonSet.propTypes = propTypes;
AudioButtonSet.defaultProps = defaultProps;

export default AudioButtonSet;