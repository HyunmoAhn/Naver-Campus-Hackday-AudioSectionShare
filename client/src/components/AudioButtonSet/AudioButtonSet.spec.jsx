import React from 'react';
import { shallow } from 'enzyme';

import AudioButtonSet from './';

describe('<AudioButtonSet />', () => {
	it('should match snapshot when render default', () => {
		const wrapper = shallow(<AudioButtonSet />);

		expect(wrapper).toMatchSnapshot();
	});

	it('should match snapshot when props.isSelectionLoop === true', () => {
		const wrapper = shallow(<AudioButtonSet
			isSectionLoop
		/>);

		expect(wrapper).toMatchSnapshot();
	})
});