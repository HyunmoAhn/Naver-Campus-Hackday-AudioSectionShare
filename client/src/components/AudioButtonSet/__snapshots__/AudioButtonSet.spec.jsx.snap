// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AudioButtonSet /> should match snapshot when props.isSelectionLoop === true 1`] = `
<div
  className="AudioButtonSet"
>
  <button
    className="AudioButtonSet__section-loop-cancel"
    onClick={[Function]}
    type="button"
  >
    <TimeScreen
      className=""
      second={0}
    />
    ~
    <TimeScreen
      className=""
      second={0}
    />
    <br />
    반복 해제
  </button>
  <button
    className="AudioButtonSet__section-btn"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    <i
      className="fa fa-exchange"
    />
  </button>
  <button
    className="AudioButtonSet__play-btn"
    onClick={[Function]}
    type="button"
  >
    <i
      className="fa fa-pause"
    />
  </button>
  <button
    className="AudioButtonSet__toggle-btn"
    onClick={[Function]}
    type="button"
  >
    <i
      className="fa fa-retweet"
    />
  </button>
  <button
    className="AudioButtonSet__share-btn"
    onClick={[Function]}
    type="button"
  >
    <span
      className=""
    >
      SNS 공유하기
    </span>
  </button>
</div>
`;

exports[`<AudioButtonSet /> should match snapshot when render default 1`] = `
<div
  className="AudioButtonSet"
>
  <button
    className="AudioButtonSet__section-btn"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <i
      className="fa fa-exchange"
    />
  </button>
  <button
    className="AudioButtonSet__play-btn"
    onClick={[Function]}
    type="button"
  >
    <i
      className="fa fa-pause"
    />
  </button>
  <button
    className="AudioButtonSet__toggle-btn"
    onClick={[Function]}
    type="button"
  >
    <i
      className="fa fa-retweet"
    />
  </button>
  <button
    className="AudioButtonSet__share-btn"
    onClick={[Function]}
    type="button"
  >
    <span
      className=""
    >
      SNS 공유하기
    </span>
  </button>
</div>
`;
